### PR TITLE
fix(send_text): wait_for_tx confirmed against evidence that can exist (was: false on delivered messages)

### PR DESCRIPTION
**Pull request by jamesarich**
## Problem (hardware-reproduced)

`send_text(wait_for_tx=True)` returns `tx_confirmed: false` for messages that **are** delivered.

Reproduced on a T-Beam S3 → Heltec V3 pair: the peer decoded the broadcast (packet id `4041547107` matching the sent id, SNR 5.5 / RSSI −45) while the sender reported failure. An agent trusting `tx_confirmed` would conclude a healthy mesh was broken.

## Root cause

The check polled the recorder's **packet** stream for the packet it had just sent:

```python
window = log_query.packets_window(start="-1m", portnum="TEXT_MESSAGE_APP", max=50)
```

That stream is fed by the `meshtastic.receive` pubsub topic (`recorder/recorder.py:145`), which carries only packets the node **received**. A node's own transmission is never published there, so the confirmation could never succeed for locally-originated traffic — a structural false negative, not a timing issue.

Confirmed on hardware: **zero** `TEXT_MESSAGE_APP` packets in the recorder across four sends, one of which a second node provably received.

## Fix

The firmware logs the transmission, with the packet id in lowercase hex:

```
enqueue for send (id=0x8738a6d9 ...)   <- queued, may still never key up
Started Tx (id=0x8738a6d9 ...)         <- the radio actually transmitted
```

`_confirm_tx` matches `Started Tx`, plus a neighbour's rebroadcast arriving back with our id (also proof it reached the air, and works with no log capture).

Deliberately **not** `enqueue for send`: a packet can be enqueued and then killed before airtime — precisely the bug fixed in #18 — so matching the enqueue line would trade a false negative for a false positive.

### `tx_confirmed` is now three-valued

Absence of evidence is not evidence of failure. The log path only reaches the recorder when `set_debug_log_api(True)` is on (or a `serial_session` taps the port).

| value | meaning |
| --- | --- |
| `true` | transmission observed |
| `false` | logs were flowing and showed no TX — a real failure signal |
| `null` | not observable — **not** a failure; `tx_unconfirmed_reason` points at `set_debug_log_api(True)` |

Also drops the old `packet_id is None` fallback that confirmed on *any* recent TEXT packet — it could match an unrelated message from another node.

## Docs

`AGENTS.md` documented the same broken pattern (`packets_window` to confirm a local send). Corrected in the workflow section and added to the anti-patterns list, since that guidance actively leads agents into this trap.

## Validation

Hardware, on the same T-Beam that reported `false`:

- `tx_confirmed=True`, latency 0.07 s
- `True` on a second independent node (Heltec V3)
- `null` + the guidance reason when no logs are captured

7 new unit tests covering all three states, both evidence sources, the enqueue-is-not-confirmation case, and the wrong-packet-id case. 29 passed across `test_wait_for_tx` / `test_mcp_surface` / `test_tool_annotations` / `test_linger_tx_race`. ruff + mypy clean.

Found while exercising the MCP after #18 merged.

## Summary by CodeRabbit

- **New Features**
  - Improved “wait for transmission” confirmation by validating over-the-air transmission evidence.
  - Transmission status is now tri-state: confirmed, not confirmed, or unknown (with an explanatory reason when unknown).

- **Documentation**
  - Rewrote “send a message and confirm delivery” guidance, including correct waiting/timeout practices and interpretation of confirmation/ack signals.
  - Clarified why local sends don’t appear as observable receive/packet-window evidence.

- **Tests**
  - Added unit tests covering confirmed/not-confirmed/unknown cases and edge conditions (e.g., log truncation and port scoping).

### CODE REVIEW

**Comment by jamesarich** [comment]
@coderabbitai review

File: src/meshtastic_mcp/server.py [hunk 1]
```diff
@@ -1288,6 +1289,72 @@ def config_diff(name_a: str, name_b: str | None = None, port: str | None = None)
...
+        if logs.get("lines"):
+            saw_any_log = True
+            for entry in logs["lines"]:
+                if started_tx.search(entry.get("line", "")):
+                    return True, round(time.monotonic() - t0, 2), None
+
+        packets = log_query.packets_window(start="-2m", max=200)
+        for pkt in packets.get("packets", []):
+            if pkt.get("id") == packet_id:
+                return True, round(time.monotonic() - t0, 2), None
```
[the remark this answers is not in the document]
Comment (jamesarich) [reply src/meshtastic_mcp/server.py]: Good catch on the premise — the numbers back it up. An unfiltered 2 min window on the bench T-Beam measured `total_matched=455 / dropped=395`, so this was a live failure mode, not a theoretical one.

I took a different fix than the suggested bisect, though, and I think it is strictly better here.

Bisecting is the right guidance for *exploratory* windowed queries, where you want the complete set. This lookup is the opposite shape: it needs exactly one known line. `log_query.logs_window` applies `grep` **before** the `max_lines` cap (`log_query.py:158-163` — `grep_re.search(...)` gates the `matched += 1` and the `out.pop(0)` trim), so pushing the packet-id pattern down as the filter collapses the matched set to ~1 row. Truncation then becomes structurally impossible for the line we care about, rather than something we detect and recover from — and it costs one filtered scan instead of recursive re-querying.

```python
tx_pattern = rf"Started Tx \(id=0x{packet_id:08x}\b"
hits = log_query.logs_window(start="-2m", port=port, grep=tx_pattern, max_lines=20)
```

`packets_window` is the genuinely exposed one — it has no `grep` (only portnum/from_node/to_node), so the rebroadcast check now honours `dropped`: if nothing matched, no log channel exists, **and** the packet window was truncated, the reason says so instead of implying a clean observation. Worth noting that path can only downgrade a `true` to a `null`, never manufacture a false negative — if the log channel is up and shows no `Started Tx`, our node did not transmit, so no rebroadcast could exist to be dropped.

The observability probe ("is any log flowing at all?") is now a separate `max_lines=1` existence query, so the cap cannot skew it either.

Added two tests: one buries the `Started Tx` line under 1000 unrelated log lines (the test double now emulates grep-before-cap, so it genuinely exercises the ordering rather than passing regardless), and one covers the truncated-packet-window reason. 9 passing, ruff + mypy clean, and re-verified on hardware — `tx_confirmed=True`, latency 0.06 s.

Fixed in c703ed3.

**Comment by jamesarich** [comment]
@coderabbitai review

File: tests/unit/test_wait_for_tx.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        makes `test_tx_line_survives_a_chatty_log` meaningful — without it the
+        test would pass no matter how the lookup was implemented."""
+        grep = kwargs.get("grep")
+        max_lines = kwargs.get("max_lines", 200)
+        lines = state["log_lines"]
+        if grep is not None:
+            import re
+
+            rx = re.compile(grep)
+            lines = [line for line in lines if rx.search(line["line"])]
```
[the remark this answers is not in the document]
Comment (jamesarich) [reply tests/unit/test_wait_for_tx.py]: Valid — fixed in 898dad6.

You are right that nothing pinned the contract: `fake_logs_window` took `**kwargs` and never read `port`, so dropping the scoping from `_confirm_tx` would have left every test green.

Worth noting it is not a theoretical gap either. Packet ids are 32 bits and a bench commonly has several radios on one channel — the one this was developed against currently has three nodes sharing `olm3sh`, so an unscoped query really could confirm a send off another radio's logs.

The double now filters by `port` and records every port queried, and there is a new test asserting that a `Started Tx` carrying our id but logged against a *different* port does not confirm, and that the target port was passed down:

```python
def test_tx_on_another_port_does_not_confirm(stub_query):
    stub_query["log_lines"] = [
        _log_line("Started Tx (id=0x8738a6d9 ...)", port="/dev/ttyUSB0"),
    ]
    confirmed, _latency, _reason = stub_query["server"]._confirm_tx(PID, "/dev/ttyACM1", 0.5)
    assert confirmed is not True
    assert "/dev/ttyACM1" in stub_query["ports_queried"]
```

On the ast-grep ReDoS warning in the same hunk: not applicable, so I left it. The pattern is not caller-controlled — `_confirm_tx` builds it from an int packet id via `f"Started Tx \\(id=0x{packet_id:08x}\\b"`, so no untrusted text reaches `re.compile`.

10 passing, ruff + mypy clean.
